cassandra/connection.py

```diff
@@ -732,3 +743,90 @@ def deliver(self, timeout=None):
             raise OperationTimedOut()
         else:
             return self.responses
+
+
+class HeartbeatFuture(object):
+    def __init__(self, connection, owner):
+        self._exception = None
+        self._event = Event()
+        self.connection = connection
+        self.owner = owner
+        log.debug("Sending options message heartbeat on idle connection %s %s",
+                  id(connection), connection.host)
+        with connection.lock:
+            connection.send_msg(OptionsMessage(), connection.get_request_id(), self._options_callback)
+            connection.in_flight += 1
+
+    def wait(self, timeout):
+        if self._event.wait(timeout):
```

Comment:
Same note on Event.wait() always returning None in Python 2.6.  See ResponseWaiter.deliver() for an example of handling this.